### PR TITLE
Platform language

### PR DESCRIPTION
**Pull request by chrisdugne**
## Detailed purpose of the PR

- removed use of `translation.getLanguage` (=phone lang) for all actions/services
- fetching `platform.lang` after login
- using this lang dynamically within datalayer
- using `translation.setLanguage` to use this lang for all components

### Testing strategy

- [x] I ensured my code is tested: 
- [x] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

**Comment by chrisdugne** [comment]
[ETQU, si la langue de mon téléphone n'est pas supportée par ma plateforme je souhaite que mon app soit dans la langue de ma plateforme](https://trello.com/c/VcEGhI2a/1382-etqu-si-la-langue-de-mon-t%C3%A9l%C3%A9phone-nest-pas-support%C3%A9e-par-ma-plateforme-je-souhaite-que-mon-app-soit-dans-la-langue-de-ma-platef)

File: src/components/correction.js [hunk 1]
```diff
@@ -195,7 +197,7 @@ class Correction extends React.PureComponent<Props> {
     index: number,
     animationStyle: AnimationStyleProp
   ) => {
-    const {answers, userAnswers, question, tip, keyPoint, isCorrect} = this.props;
+    const {answers, userAnswers, question, tip, keyPoint, isCorrect, language} = this.props;
```
Comment (VincentCATILLON) [inline src/components/correction.js]: IMO a component should not handle language, it should be returned by a generic function used by the entire app (implementation doesn't matter)

File: src/components/lesson.js [hunk 2]
```diff
@@ -83,11 +85,7 @@ class Lesson extends React.Component<Props> {
             openedResource &&
             openedResource.subtitleRef &&
             // @todo use user language
-            getSubtitlesUri(
-              brandTheme.host,
-              openedResource.subtitleRef,
-              translations.getLanguage()
-            );
+            getSubtitlesUri(brandTheme.host, openedResource.subtitleRef, language);
```
Comment (VincentCATILLON) [inline src/components/lesson.js]: same for this

File: src/containers/tab-bar-slide.js [hunk 3]
```diff
@@ -140,12 +141,15 @@ class TabBarSlide extends React.Component<Props> {
...
-    const labelText = getLabelText(scene);
+    const labelKey = getLabelText(scene);
 
-    if (!labelText) {
+    if (!labelKey) {
       return null;
     }
 
+    // $FlowFixMe Cannot access computed property
+    const labelText = translations[labelKey];
```
Comment (VincentCATILLON) [inline src/containers/tab-bar-slide.js]: don't understand why this function should change?

File: src/utils/local-token.test.js [hunk 4]
```diff
@@ -12,8 +12,17 @@ describe('LocalToken', () => {
...
   });
+  it('should return remove a token', async () => {
+    AsyncStorage.getItem = jest.fn().mockImplementation(() => Promise.resolve(null));
+    await localToken.set('mytoken');
+    const token = localToken.get();
+    expect(token).resolves.toEqual('mytoken');
+    await localToken.remove();
+    const token2 = localToken.get();
+    return expect(token2).resolves.toEqual(null);
+  });
```
Comment (VincentCATILLON) [inline src/utils/local-token.test.js]: this test doesn't test it correctly

**Pull request merged**